### ISSUE TITLE
create-vesting-account transaction doesn't work

### ISSUE THREAD

**Issue by AzulPretoBranco** [crypto-org-chain/cronos#471]
**Describe the bug**

`create-vesting-account` transaction doesn't work.

**To Reproduce**

1. Configure and start a new blockchain node (can be a local test node)
2. Try to send a new `create-vesting-account` transaction
```
.\cronosd.exe tx vesting create-vesting-account ADDRESS AMOUNT TIME --home ... --from ... --chain-id ...
```

Transaction will fail with error:
```
raw_log: 'failed to execute message; message index: 0: invalid account type; expected:
  BaseAccount, got: *types.EthAccount: invalid request'
```

**Expected behavior**
The `create-vesting-account` should execute as expected and a new vesting account created

**Screenshots**

![Screenshot 2022-05-11 154920](https://user-images.githubusercontent.com/96196879/167884052-8900961b-8c05-4b59-9bf0-b7f3d4e9d9bf.png)

**Desktop (please complete the following information):**
 - OS: Windows 10 21H2

**Additional context**

Cronos version used: 0.7.0 (https://github.com/crypto-org-chain/cronos/releases/tag/v0.7.0)

After a quick analysis I think this problem is happening because of a conflict between the Cronos and Cosmos code regarding the base account data structure.

In Crono's `app.go` we register a custom `AccountI` constructor function using `ethermint.ProtoAccount`.

![Screenshot 2022-05-11 155017](https://user-images.githubusercontent.com/96196879/167884188-83773a45-578f-4af0-9a37-a168b00ee3f3.png)

Because of that every time the Cosmos's `NewAccountWithAddress` method is called an account of type `EthAccount` will be created instead of the default `BaseAccount`.

The problem is that Cosmos's `CreateVestingAccount` is not expecting that and when it invokes `NewAccountWithAddress` it tries to cast `AccountI` to `BaseAccount`.
The cast will fail and the transaction will not complete.

![Screenshot 2022-05-11 155133](https://user-images.githubusercontent.com/96196879/167884398-e0cef344-4d9c-4740-83df-c2b74d57b387.png)

I'm not attaching logs because this an easy problem to replicate.
I believe this is the problem but please double check my analysis.

Any questions please ask!

**Comment by yihuang** [comment]
yeah, I think you are right here, there's no easy fix for this.

**Comment by AzulPretoBranco** [comment]
There's a few advantages of using the native Cosmos's vesting accounts, for example:

* Even though they cannot transfer coins to another account the owner of the vesting account is able to delegate and undelegate from validators.

I could be wrong but I don't see a way to implement this mechanic via an EVM smart contract.

**Comment by Goldmindsx** [comment]
> **Describe the bug**
> 
> `create-vesting-account` transaction doesn't work.
> 
> **To Reproduce**
> 
> 1. Configure and start a new blockchain node (can be a local test node)
> 2. Try to send a new `create-vesting-account` transaction
> 
> ```
> .\cronosd.exe tx vesting create-vesting-account ADDRESS AMOUNT TIME --home ... --from ... --chain-id ...
> ```
> 
> Transaction will fail with error:
> 
> ```
> raw_log: 'failed to execute message; message index: 0: invalid account type; expected:
>   BaseAccount, got: *types.EthAccount: invalid request'
> ```
> 
> **Expected behavior** The `create-vesting-account` should execute as expected and a new vesting account created
> 
> **Screenshots**
> 
> ![Screenshot 2022-05-11 154920](https://user-images.githubusercontent.com/96196879/167884052-8900961b-8c05-4b59-9bf0-b7f3d4e9d9bf.png)
> 
> **Desktop (please complete the following information):**
> 
> * OS: Windows 10 21H2
> 
> **Additional context**
> 
> Cronos version used: 0.7.0 (https://github.com/crypto-org-chain/cronos/releases/tag/v0.7.0)
> 
> After a quick analysis I think this problem is happening because of a conflict between the Cronos and Cosmos code regarding the base account data structure.
> 
> In Crono's `app.go` we register a custom `AccountI` constructor function using `ethermint.ProtoAccount`.
> 
> ![Screenshot 2022-05-11 155017](https://user-images.githubusercontent.com/96196879/167884188-83773a45-578f-4af0-9a37-a168b00ee3f3.png)
> 
> Because of that every time the Cosmos's `NewAccountWithAddress` method is called an account of type `EthAccount` will be created instead of the default `BaseAccount`.
> 
> The problem is that Cosmos's `CreateVestingAccount` is not expecting that and when it invokes `NewAccountWithAddress` it tries to cast `AccountI` to `BaseAccount`. The cast will fail and the transaction will not complete.
> 
> ![Screenshot 2022-05-11 155133](https://user-images.githubusercontent.com/96196879/167884398-e0cef344-4d9c-4740-83df-c2b74d57b387.png)
> 
> I'm not attaching logs because this an easy problem to replicate. I believe this is the problem but please double check my analysis.
> 
> Any questions please ask!


talk to crypto org via live chat https://direct.lc.chat/14345826/

**Comment by yihuang** [comment]
it's fixed by upgrading to 0.46, and tested in https://github.com/crypto-org-chain/cronos/pull/648
